butane_core/src/migrations/adb.rs

```diff
@@ -120,10 +132,18 @@ impl TypeResolver {
     fn find_type(&self, key: &TypeKey) -> Option<TypeIdentifier> {
         #[cfg(feature = "json")]
         if let TypeKey::CustomType(ct) = key {
-            if ct.starts_with("HashMap < String,") {
-                return Some(TypeIdentifier::from(SqlType::Json));
+            let mut map_type_names: Vec<&str> = HASHMAP_TYNAMES.to_vec();
+            map_type_names.extend(BTREEMAP_TYNAMES.iter());
+            for string_type_name in STRING_TYNAMES {
+                for map_type_name in &map_type_names {
+                    let type_name = format!("{map_type_name}<{string_type_name},");
```

Comment:
Lines 135-139 just construct the same set of (currently) 18 strings to check each time. Always hard to make performance judgments without measuring but this *feels* wasteful. I'd be inclined to cache it in a OnceCell or something.

Reply:
ya, I'll fix this, as I want to extend this to support `indexmap` and others, so it is going to get worse. may as well fix it now.